src/Symfony/Component/DependencyInjection/EnvVarProcessor.php

```diff
@@ -366,4 +369,10 @@ public function getEnv(string $prefix, string $name, \Closure $getEnv): mixed
 
         throw new RuntimeException(sprintf('Unsupported env var prefix "%s" for env name "%s".', $prefix, $name));
     }
+
+    public function reset(): void
+    {
+        $this->loadedVars = [];
+        $this->loaders = $this->originalLoaders;
```

Comment:
This looks useless to me. The only place where we mutate the loaders property already resets it. We don't need this `originalLoaders` property which will always contain the same iterator here at this point.

Reply:
There still this so it might be better to do have this reset?
https://github.com/symfony/symfony/blob/4452c58f0ab1fd5b36e91e2153b5832b5c544187/src/Symfony/Component/DependencyInjection/EnvVarProcessor.php#L186-L187

Reply:
In theory, the iterator might not be rewindable, but let's ignore this concern until someone reports it creates an issue for them :)

Reply:
$loaders variable is overridden here again, and then this->loaders is mutated in finally section 

https://github.com/symfony/symfony/blob/4452c58f0ab1fd5b36e91e2153b5832b5c544187/src%2FSymfony%2FComponent%2FDependencyInjection%2FEnvVarProcessor.php#L210

The second test I wrote confirms it, without the original loader thingy, env vars are set to '' on subsequent calls

Reply:
@stof